.github/mergify.yml

```
@@ -58,26 +143,313 @@ pull_request_rules:
+          Feel free to reopen it if you think it's a mistake.
+
+  - name: remove certain labels on close
+    conditions:
+      - closed
+    actions:
+      label:
+        remove:
+          - triage
+
+  - name: "close Pull Requests without files changed"
+    conditions:
+      - and: *is_open
+      - "#files=0"
+    actions:
+      label:
+        add:
+          - "no-changes"
+      close:
+        message: "This pull request has been automatically closed because there are no longer any changes."
+
+  - name: welcome new contributors
+    conditions:
+      - and: *is_open
+      - and: *not_a_bot
+    actions:
+      comment:
+        message: |
+          Thanks @{{author}} for creating this pull request! 
+          A maintainer will review your changes shortly.
+          Please don't be discouraged if it takes a while.
+
+          While you wait, make sure to review our
+          [contributor guidelines](https://github.com/cloudposse/.github/blob/main/CONTRIBUTING.md).
+
+          > [!TIP]
+          > ## Need help or want to ask for a PR review to be expedited?
+          > Join us on [Slack](https://slack.cloudposse.com) in the `#pr-reviews` channel.
+
+  - name: add triage label for new pull requests
+    conditions:
+      - and: *is_open
+      - '#label=0'
+      - or:
+        - created-at > 5 minutes ago
+        - commits[*].date_committer > 5 minutes ago
+        - updated-at > 7 days ago
+    actions:
+      label:
+        add:
+          - triage
+
+  - name: Add needs-test label on new commits
+    conditions:
+      - and: *is_open
+      - or: *default_branch
+      - and: *is_terraform
+      - and: *is_recent_commit
+      - -label=~needs-test
+    actions:
+      label:
+        add: ["needs-test"]
+
+  - name: Remove needs-test label when required tests pass
+    conditions:
+      - and: *is_open
+      - or: *default_branch
+      - and: *require_terraform_checks_passing
+    actions:
+      label:
+        remove: ["needs-test"]
+
+  - name: add "WIP" label when the title contains "WIP"
+    conditions:
+      - and: *is_open
+      - title~=WIP
+    actions:
+      label:
+        toggle:
+          - wip
+
+  - name: add "needs-cloudposse" label when restrictions apply to this PR
+    conditions:
+      - and: *is_open
+      - and: *needs_cloudposse
+    actions:
+      label:
+        toggle:
+          - needs-cloudposse
+      comment:
+        message: |
+          > [!IMPORTANT]
+          > This pull request modifies files that require Cloud Posse's review.
+          > Please be patient, and a core maintainer will review your changes.
+          > To expedite this process, reach out to us on [Slack](https://slack.cloudposse.com)
+          > in the `#pr-reviews` channel.
+
+  - name: rebase pull request when it's more than 10 commits behind main
+    conditions:
+      - and: *is_open
+      - or: *default_branch
+      - "#commits-behind>=10"
```

tbh it should probably be 1 due to all the automation in place

```suggestion
      - "#commits-behind>=1"
```